lib/logstash/codecs/compress_spooler.rb

```diff
@@ -21,13 +21,13 @@ def decode(data)
   end # def decode
 
   def encode(event)
+    # use normalize to make sure returned Hash is pure Ruby for
+    # MessagePack#pack which relies on pure Ruby object recognition
+    @buffer << LogStash::Util.normalize(event.to_hash).merge(LogStash::Event::TIMESTAMP => event.timestamp.to_iso8601)
+    # If necessary, we flush the buffer and get the data compressed
```

Comment:
but you are making a copy using merge `merge(LogStash::Event::TIMESTAMP => event.timestamp.to_iso8601)` in this case - so it was a more general observation.